src/components/trade/LastTrades/LastTradesWidget.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { TokenDetails } from 'types'
+import { LastTrades, LastTradesItem } from './LastTrades'
+
+import { useQuery, gql } from '@apollo/client'
+import BigNumber from 'bignumber.js'
+import { calculatePrice } from '@gnosis.pm/dex-js'
+
+const GET_TRADES = gql`
+  query GetTrades {
+    trades(first: 10, orderBy: tradeBatchId, orderDirection: desc, where: { revertEpoch: null }) {
+      id
+      sellVolume
+      buyVolume
+      tradeEpoch
+      sellToken {
+        decimals
+      }
+      buyToken {
+        decimals
+      }
+    }
+  }
+`
+
+export interface Props {
+  quoteToken: TokenDetails
+}
+
+function toLastTradeItem(trade: any): LastTradesItem {
+  const { id, sellVolume, buyVolume, tradeEpoch, sellToken, buyToken } = trade
+  const sellTokenDecimals = +sellToken.decimals
+  return {
+    id,
+    price: calculatePrice({
+      denominator: { amount: sellVolume, decimals: sellTokenDecimals },
+      numerator: { amount: buyVolume, decimals: +buyToken.decimals },
+    }),
+    size: new BigNumber(sellVolume).div(new BigNumber(10).pow(18 - sellTokenDecimals)),
+    time: new Date(Number(tradeEpoch) * 1000),
+  }
+}
+
+export const LastTradesWidget: React.FC<Props> = (props) => {
+  const { loading, error, data } = useQuery(GET_TRADES)
+
+  const { quoteToken } = props
+  const lastTrades: LastTradesItem[] = data ? data.trades.map(toLastTradeItem) : []
```

Comment:
memoize?

Reply:
Sure, I can memoize, but is it worth it at this point for pure data and given how simple tries to be this component?

![image](https://user-images.githubusercontent.com/2352112/93114027-57772f00-f6ba-11ea-91ad-873a817bab6f.png)